wandb/sdk/internal/sender.py

```diff
@@ -821,7 +821,7 @@ def _maybe_setup_resume(
         # TODO: Do we need to restore config / summary?
         # System metrics runtime is usually greater than history
         self._resume_state.runtime = max(events_rt, history_rt)
-        self._resume_state.step = history.get("_step", -1) + 1 if history else 0
+        self._resume_state.step = history.get("_step", -1) + 1 if history and bool(set(summary.keys()) - {"_runtime", "_timestamp", "_step", "_wandb"}) else 0
```

Comment:
this is on the right track, but what do you think about something like this:
```suggestion
        last_step = history.get("_step", 0)
        history_line_count = resume_status["historyLineCount"]
        self._resume_state.step = last_step + 1 if history_line_count > 0 else last_step
```
i'm thinking in case the step is not defined we will use 0 as default init value. 
Then we check if we already logged anything:
-  in case we didn't we going to use the last step 
- otherwise we increment the last step by 1.
note that we are using resume_status["historyLineCount"] in the next line, so we can reuse the variable.

---
There is a follow up task here to figure out what to do in case first step is negative (currently we will drop it), but it is outside the scope of this task, just wanted to let you know. 
```
run.log({"loss":0.01}, step=-1, commit=True)
```
we will drop this one since we init internal step at zero.

Reply:
Didn't realize about `resume_status["historyLineCount"]`, it's definitely cleaner!